state-chain/pallets/cf-lp/src/lib.rs

```diff
@@ -319,9 +311,11 @@ pub mod pallet {
 				Error::<T>::BoostedFundsRemaining
 			);
 
-			let _ = FreeBalances::<T>::clear_prefix(&account_id, u32::MAX, None);
+			// Clear all liquidity refund addresses for the account.
 			let _ = LiquidityRefundAddress::<T>::clear_prefix(&account_id, u32::MAX, None);
-			let _ = HistoricalEarnedFees::<T>::clear_prefix(&account_id, u32::MAX, None);
```

Comment:
Indeed, that's a bit problematic 😅. Ill check where it could make sense to add without reintroducing a dependency between pallets.

Reply:
I think there are two options and both include a cross dependency between pallets:

1) Keep it now in the pool's pallet and add a trait with a method to delete the storage
2) Move it back in the LP pallet with a trait to call from the pool pallet to record historical fees

A third option would be to put it to the `on_killed_account` method like you proposed, but that's a bit hacky in my opinion because the account is actually not killed, just unregistered. I am not 100% sure if we really want to remember the historical fees even if someone unregisters his account as LP. That could be seen as a bug or a feature, not sure though. What's your opinion?

Reply:
Sorry, late reply. As discussed - I think using the OnKilledAccount hook is fine.